docs/rules/no-invalid-at-rule-placement.md

```diff
@@ -0,0 +1,95 @@
+# no-invalid-at-rule-placement
+
+Disallow invalid placement of at-rules.
+
+## Background
+
+At-rules are CSS statements that instruct CSS how to behave. Some at-rules have strict placement requirements that must be followed for the stylesheet to work correctly. For example:
```

Comment:
`@charset` is [no longer considered an at-rule](https://developer.mozilla.org/en-US/docs/Web/CSS/@charset) even though it syntactically looks like one. There may be a more suitable name and wording in the documentation that better reflect this.

Reply:
If it's not an at-rule, then what is it? The spec is ridiculously worded:
https://drafts.csswg.org/css-syntax/#charset-rule

Reply:
Interestingly, I think this is arguably valid even without proper uses of `@charset` being at-rules. From the docs:

> If a `@charset` is not recognized as the charset declaration, it is parsed as a normal at-rule. 

...so an improper `@charset` is roughly speaking, an invalid at-rule placement. Sort of.